sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/intervalExpressions.scala

```diff
@@ -615,15 +615,19 @@ case class DivideYMInterval(
   @transient
   private lazy val evalFunc: (Int, Any) => Any = right.dataType match {
     case LongType => (months: Int, num) =>
+      if (num == 0) throw QueryExecutionErrors.divideByZeroError()
```

Comment:
We don't need to double check proactively that `num` is 0. `LongMath.divide` will do that for you. Just catch the Java exception, and replace it by Spark's one.

Reply:
`try ... catch ...` looks less elegant than direct judgement.

Reply:
hmm, I cannot say that duplicating checks per every case looks so elegant too.

Reply:
performance-wise, I think try-catch is better for the common case (no error).